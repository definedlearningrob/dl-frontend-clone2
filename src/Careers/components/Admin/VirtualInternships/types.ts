import { TOpportunity as TBaseOpportunity } from '@dc/resources/types';

export const enum VirtualInternshipStatuses {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export type VirtualInternshipLessonAttributes = {
  lessonId: string;
  step: number;
};

export type FormValues = {
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  description: string;
  name: string;
  pathways: {
    label: string;
    value: string;
  }[];
  availableSpots: number;
  creditsOutcomes: string;
  image: Record<string, unknown>;
  tags: {
    label: string;
    value: string;
  }[];
  requiredExperiences: number;
  status: { value: VirtualInternshipStatuses; label: string };
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};

export type TPathway = Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>;

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'opportunityType'
  | 'tags'
  | 'salaryInformation'
> & {
  pathways: TPathway[];
};

export type Lesson = {
  id: string;
  step: number;
  name: string;
  imageUrl: string;
  thumbnailUrl: string;
  type: string;
};

export type TVirtualInternship = {
  archivedAt: string | null;
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  id: string;
  opportunity: TOpportunity;
  requiredExperiences: number;
  status: VirtualInternshipStatuses;
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};
