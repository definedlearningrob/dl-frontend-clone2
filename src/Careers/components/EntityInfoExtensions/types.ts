export type TAdminEntityInfoExtensionField = {
  id: string;
  name: string;
};
