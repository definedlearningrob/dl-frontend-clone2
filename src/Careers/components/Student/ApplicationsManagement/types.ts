export type Counselor = {
  email: string;
  firstName: string;
  fullName: string;
  lastName: string;
  name: string;
  username: string;
  uuid: string;
};

export type AvailableTeacher = {
  email: string;
  fullName: string;
  name: string;
  username: string;
  uuid: string;
};

export type CounselorSelectOption = { label: string; value: string };
