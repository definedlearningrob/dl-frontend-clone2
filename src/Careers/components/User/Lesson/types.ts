import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

import {
  TCheckInGroup as TBaseCheckInGroup,
  TCheckInQuestion as TBaseCheckInQuestion,
} from '@shared/components/CheckIns/types';

type TRubricHeading = {
  id: string;
  multiplier: number;
  name: string;
  uuid: string;
};

type TRubricCriteria = {
  id: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
  uuid: string;
};

type TRubricCriteriaLabel = {
  displayName: string;
  id: string;
  score: number;
  uuid: string;
};

type TRubric = {
  id: string;
  uuid: string;
  name: string;
  description: string;
  headings: TRubricHeading[];
  criterias: TRubricCriteria[];
  criteriaLabels: TRubricCriteriaLabel[];
};

type TAssignmentSubmissionFile = {
  filename: string;
  googleWeblink?: string;
  id: string;
  source: string;
  url: string;
};

type TSubmissionGrade = {
  id: string;
  lastGradedBy: {
    uuid: string;
    firstName: string;
    lastName: string;
    fullName: string;
  };
  status: 'ACCEPTED' | 'NOT_ACCEPTED';
  createdAt: string;
  updatedAt: string;
};

export type TAssignmentSubmission = {
  files: TAssignmentSubmissionFile[];
  grade: TSubmissionGrade | null;
  id: string;
  status: 'DRAFT' | 'SUBMITTED';
};

type TAttachmentFile = {
  filename: string;
  id: string;
  url: string;
};

type TExtensionField = {
  archivedAt: string | null;
  author: {
    email: string;
    firstName: string;
    lastName: string;
    username: string;
    uuid: string;
  };
  description: string;
  files: {
    filename: string;
    id: string;
    url: string;
  }[];
  id: string;
  imageUrl: string;
  links: {
    name: string;
    url: string;
  }[];
  name: string;
};

type TBadge = {
  archivedAt: string | null;
  id: string;
  imageUrl: string;
  name: string;
};

export type TAssignment = {
  archivedAt: string | null;
  assetName: string;
  badges: TBadge[];
  createdAt: string;
  id: string;
  name: string;
  description: string;
  displayName: string;
  step: number;
  rubrics: TRubric[];
  submission: TAssignmentSubmission | null;
  lessons: TLesson[];
  __typename: 'Assignment';
};

export type TAttachment = {
  archivedAt: string | null;
  createdAt: string;
  description: string;
  displayName: string;
  files: TAttachmentFile[];
  id: string;
  step: number;
  lessons: TLesson[];
  name: string;
  __typename: 'Attachment';
};

export type TCareerReviewSurvey = {
  questions: {
    answer: string[];
    id: string;
    options: {
      option: string;
      step: number;
    }[];
    question: string;
    type: SURVEY_QUESTION_TYPE;
  }[];
};

export type TExternalPresentation = {
  archivedAt: string | null;
  displayName: string;
  id: string;
  source: string;
  step: number;
  isExpandable: boolean;
  __typename: 'ExternalPresentation';
};

export type TResearchLink = {
  archivedAt: string | null;
  author: string;
  createdAt: string;
  displayName: string;
  id: string;
  resourceLink: string;
  sourceName: string;
  step: number;
  __typename: 'ResearchLink';
};

export type TText = {
  archivedAt: string | null;
  content: string;
  createdAt: string;
  displayName: string;
  id: string;
  step: number;
  __typename: 'Text';
};

export type TVideo = {
  archivedAt: string | null;
  createdAt: string;
  description: string;
  displayName: string;
  filename: string;
  id: string;
  step: number;
  url: string;
  __typename: 'Video';
};

export type TVocabulary = {
  archivedAt: string | null;
  createdAt: string;
  definition: string;
  id: string;
  step: number;
  term: string;
  __typename: 'Vocabulary';
};

export type TCheckInQuestion = Pick<
  TBaseCheckInQuestion,
  'id' | 'question' | 'step' | 'answer' | '__typename'
>;

export type TCheckInGroup = Omit<TBaseCheckInGroup, 'questions'> & {
  questions: TCheckInQuestion[];
};

export type TLesson = {
  archivedAt: string | null;
  id: string;
  imageUrl: string;
  name: string;
  assignments: TAssignment[];
  attachments: TAttachment[];
  badges: TBadge[];
  careerReviewSurvey: TCareerReviewSurvey | null;
  checkInGroups: TCheckInGroup[];
  checkInQuestions: TCheckInQuestion[];
  description: {
    audience: string;
    goal: string;
    introduction: string;
    role: string;
    situation: string;
  };
  extensionFields: TExtensionField[];
  externalPresentations: TExternalPresentation[];
  hasPresentation: boolean;
  researchLinks: TResearchLink[];
  step: number;
  texts: TText[];
  thumbnailUrl: string;
  type: string;
  videos: TVideo[];
  vocabularies: TVocabulary[];
};
