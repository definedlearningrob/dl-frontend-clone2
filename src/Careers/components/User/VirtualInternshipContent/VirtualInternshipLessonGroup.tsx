import { useParams } from 'react-router-dom';

import { LessonCard } from '@dc/shared/LessonCard';
import { TLesson } from '@dc/resources/types';

import { cx } from '@shared/utils/cx';
import SkeletonRectangle from '@shared/components/Skeleton/Rectangle/Rectangle';

type Lesson = Pick<TLesson, 'id' | 'thumbnailUrl' | 'type' | 'name'>;

type Props = {
  title: string;
  description: string;
  lessons: Lesson[] | undefined;
  centerItems?: boolean;
  loading?: boolean;
};

export const VirtualInternshipLessonGroup = ({
  title,
  description,
  lessons,
  centerItems,
  loading,
}: Props) => {
  const { opportunityId, virtualInternshipId } =
    useParams<{ opportunityId: string; virtualInternshipId: string }>();

  return (
    <div aria-labelledby='lesson-group-name' data-testid='lesson-group'>
      <h5 className='text-sm xxxl:text-base mb-xxs xxxl:mb-xs' id='lesson-group-name'>
        {title}
      </h5>
      <p className='text-xs xxxl:text-sm leading-lg mb-base xxxl:mb-md'>{description}</p>
      <div
        className={cx('flex flex-wrap gap-y-base gap-x-sm xxxl:gap-y-md', {
          'justify-center': centerItems,
        })}>
        {lessons?.map((lesson) => (
          <LessonCard
            key={lesson.id}
            className='grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]'
            imageUrl={lesson.thumbnailUrl}
            lessonType={lesson.type}
            name={lesson.name}
            navigationPath={`/opportunities/${opportunityId}/virtual-internship/${virtualInternshipId}/lessons/${lesson.id}`}
            showProgress={false}
            surveyPerformed={false}
          />
        ))}
        {loading &&
          Array.from({ length: 4 }).map((_, index) => (
            <SkeletonRectangle
              key={index}
              className='grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))] !h-[230px]'
              height='extra-large'
              radius='sm'
            />
          ))}
      </div>
    </div>
  );
};
