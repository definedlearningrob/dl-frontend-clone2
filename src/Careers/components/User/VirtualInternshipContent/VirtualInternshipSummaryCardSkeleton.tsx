import SharedCard from '@shared/components/Card/Card';
import SkeletonRectangle from '@shared/components/Skeleton/Rectangle/Rectangle';

export const VirtualInternshipSummaryCardSkeleton = () => (
  <SharedCard>
    <div className='flex items-start flex-wrap gap-base xl:flex-nowrap'>
      <SkeletonRectangle
        className='grow-0 shrink-0 !h-[150px] basis-[27%] xxxl:basis-[32%]'
        radius='sm'
      />
      <div className='flex-1'>
        <SkeletonRectangle className='mb-xs' height='small' radius='sm' size='md' />
        <SkeletonRectangle className='mb-xxs' size='lg' />
        <SkeletonRectangle className='mb-xxs' size='lg' />
      </div>
    </div>
  </SharedCard>
);
