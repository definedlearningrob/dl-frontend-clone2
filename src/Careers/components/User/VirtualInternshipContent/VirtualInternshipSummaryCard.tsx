import { VirtualInternshipHeader } from '@dc/components/VirtualInternship/VirtualInternshipHeader';

import { cleanInjection } from '@shared/utils/cleanInjection';

import { VirtualInternshipSummaryCardSkeleton } from './VirtualInternshipSummaryCardSkeleton';

type Props = {
  opportunity:
    | {
        imageUrl: string;
        name: string;
        description: string;
      }
    | undefined;
};

export const VirtualInternshipSummaryCard = ({ opportunity }: Props) => {
  if (!opportunity) {
    return <VirtualInternshipSummaryCardSkeleton />;
  }

  return (
    <VirtualInternshipHeader imageUrl={opportunity.imageUrl} title={opportunity.name}>
      {/* eslint-disable-next-line react/no-danger */}
      <p dangerouslySetInnerHTML={cleanInjection(opportunity.description)} />
    </VirtualInternshipHeader>
  );
};
