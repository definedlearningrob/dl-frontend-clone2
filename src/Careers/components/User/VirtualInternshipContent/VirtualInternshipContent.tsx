import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import { isEmpty } from 'lodash-es';

import { useVirtualInternshipQuery } from '@dc/graphql/user/hooks/useVirtualInternshipQuery';

import SharedCard from '@shared/components/Card/Card';

import { VirtualInternshipLessonGroup } from './VirtualInternshipLessonGroup';
import { VirtualInternshipSummaryCard } from './VirtualInternshipSummaryCard';

export const VirtualInternshipContent = () => {
  const { t } = useTranslation();
  const { virtualInternshipId } = useParams<{ virtualInternshipId: string }>();
  const { data, loading } = useVirtualInternshipQuery({ id: virtualInternshipId, track: true });

  const virtualInternship = data?.virtualInternship;

  const hasReadinessSkillsLessons = !isEmpty(virtualInternship?.readinessSkillsLessons);

  return (
    <>
      <VirtualInternshipSummaryCard opportunity={virtualInternship?.opportunity} />
      <SharedCard className='flex flex-col gap-base xxxl:gap-md'>
        <VirtualInternshipLessonGroup
          description={t('virtualInternship.onboarding.description')}
          lessons={virtualInternship?.calendarLessons}
          loading={loading}
          title={t('virtualInternship.onboarding.heading')}
        />
        <SharedCard className='border border-neutral-300'>
          <VirtualInternshipLessonGroup
            centerItems={true}
            description={t('virtualInternship.experienceOpportunity.description', {
              lessonCount: virtualInternship?.requiredExperiences,
            })}
            lessons={virtualInternship?.experienceOpportunityLessons}
            loading={loading}
            title={t('virtualInternship.experienceOpportunity.heading')}
          />
        </SharedCard>
        <VirtualInternshipLessonGroup
          description={t('virtualInternship.postExperience.description')}
          lessons={virtualInternship?.postExperienceLessons}
          loading={loading}
          title={t('virtualInternship.postExperience.heading')}
        />
      </SharedCard>
      {hasReadinessSkillsLessons && (
        <SharedCard>
          <VirtualInternshipLessonGroup
            description={t('virtualInternship.careerReadinessSkill.description')}
            lessons={virtualInternship?.readinessSkillsLessons}
            loading={loading}
            title={t('virtualInternship.careerReadinessSkill.heading')}
          />
        </SharedCard>
      )}
    </>
  );
};
