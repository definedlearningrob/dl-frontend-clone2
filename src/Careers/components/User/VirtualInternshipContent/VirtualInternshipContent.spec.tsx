import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { Route } from 'react-router-dom';
import { screen, within } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import { OpportunityTypes } from '@graphql/dc/shared/types';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import { userInfoMock } from '@dc/tests/mocks/userMocks';
import {
  TVirtualInternshipData,
  TVirtualInternshipVariables,
  VIRTUAL_INTERNSHIP_QUERY,
} from '@dc/graphql/user/queries/virtualInternship';
import { VirtualInternshipStatuses } from '@dc/components/Admin/VirtualInternships/types';

import { VirtualInternshipContent } from './VirtualInternshipContent';

const virtualInternshipMock: MockedResponse<TVirtualInternshipData, TVirtualInternshipVariables> = {
  request: {
    query: VIRTUAL_INTERNSHIP_QUERY,
    variables: { id: '2', track: true },
  },
  result: {
    data: {
      virtualInternship: {
        archivedAt: null,
        badges: [],
        id: '2',
        opportunity: {
          id: '1',
          name: 'Test Virtual Internship',
          availableSpots: 1000000000,
          creditsOutcomes: '',
          description: 'Virtual internship description',
          imageUrl: 'https://picsum.photos/600/300',
          opportunityType: OpportunityTypes.VIRTUAL_INTERNSHIP,
          pathways: [
            {
              id: '44',
              name: 'Test pathway',
            },
          ],
          salaryInformation: null,
          tags: ['Software', 'Test'],
        },
        requiredExperiences: 2,
        status: VirtualInternshipStatuses.PUBLISHED,
        calendarLessons: [
          {
            id: '74',
            step: 3,
            name: 'Career Pathway: Maintenance/Operations',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'pathway',
          },
          {
            id: '77',
            step: 2,
            name: 'Cluster Lesson',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'career_cluster',
          },
          {
            id: '78',
            step: 1,
            name: 'Project Lesson',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'project',
          },
          {
            id: '69',
            step: 4,
            name: 'Pathway Lesson',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'pathway',
          },
        ],
        experienceOpportunityLessons: [
          {
            id: '67',
            step: 2,
            name: 'First experience opportunity',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'experience_opportunity',
          },
          {
            id: '68',
            step: 1,
            name: 'Second experience opportunity',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'experience_opportunity',
          },
        ],
        postExperienceLessons: [
          {
            id: '33',
            step: 3,
            name: 'Career Review Survey',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'career_review_survey',
          },
        ],
        readinessSkillsLessons: [
          {
            id: '58',
            step: 1,
            name: 'Readiness skills section',
            imageUrl: 'https://picsum.photos/600/300',
            thumbnailUrl: 'https://picsum.photos/600/300',
            type: 'career_readiness',
          },
        ],
      },
    },
  },
};

const renderComponent = () =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={[userInfoMock, virtualInternshipMock]}>
      <UserInfoProvider
        value={{
          userInfo: userInfoMock.result.data.userInfo,
        }}>
        <Route path='/opportunities/:opportunityId/virtual-internship/:virtualInternshipId'>
          <VirtualInternshipContent />
        </Route>
      </UserInfoProvider>
    </MockedProvider>,
    { route: '/opportunities/1/virtual-internship/2' }
  );

describe('VirtualInternshipContent', () => {
  it('renders correctly', async () => {
    const { container } = renderComponent();

    const virtualInternshipHeading = await screen.findByRole('heading', {
      name: 'Test Virtual Internship',
    });
    expect(virtualInternshipHeading).toBeInTheDocument();

    const lessonGroups = screen.getAllByTestId('lesson-group');
    expect(lessonGroups).toHaveLength(4);

    expect(within(lessonGroups[0]).getAllByRole('link')).toHaveLength(4);
    expect(within(lessonGroups[1]).getAllByRole('link')).toHaveLength(2);
    expect(within(lessonGroups[2]).getAllByRole('link')).toHaveLength(1);
    expect(within(lessonGroups[3]).getAllByRole('link')).toHaveLength(1);

    expect(container).toMatchSnapshot();
  });

  it('navigates to the lesson when clicking on lesson card', async () => {
    const { history } = renderComponent();
    history.push = jest.fn();

    const lessonCard = await screen.findByRole('link', {
      name: 'Career Pathway: Maintenance/Operations',
    });
    userEvent.click(lessonCard);

    expect(history.push).toBeCalledWith('/opportunities/1/virtual-internship/2/lessons/74');
  });
});
