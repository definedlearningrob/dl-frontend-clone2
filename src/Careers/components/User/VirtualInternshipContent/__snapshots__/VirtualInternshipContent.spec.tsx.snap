// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualInternshipContent renders correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="flex flex-wrap gap-base xl:flex-nowrap"
    >
      <div
        class="imageWrapper"
      >
        <img
          alt="Test Virtual Internship"
          class="w-full h-full object-cover"
          src="https://picsum.photos/600/300"
        />
      </div>
      <div
        class="cardContent"
      >
        <div
          class="header"
        >
          <div
            class="flex flex-col gap-sm items-start"
          >
            <h2
              class="text-font-primary mb-0 titleMedium mb-xs text-sm xxxl:text-lg"
              data-testid="card-title"
            >
              Test Virtual Internship
            </h2>
          </div>
          <div
            class="actions"
          />
        </div>
        <div
          class="description"
        >
          <p>
            Virtual internship description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card flex flex-col gap-base xxxl:gap-md"
  >
    <div
      aria-labelledby="lesson-group-name"
      data-testid="lesson-group"
    >
      <h5
        class="text-sm xxxl:text-base mb-xxs xxxl:mb-xs"
        id="lesson-group-name"
      >
        Onboarding
      </h5>
      <p
        class="text-xs xxxl:text-sm leading-lg mb-base xxxl:mb-md"
      >
        Begin exploring a career in this cluster and understand a few soft skills that are important to succeed.
      </p>
      <div
        class="flex flex-wrap gap-y-base gap-x-sm xxxl:gap-y-md"
      >
        <a
          aria-label="Career Pathway: Maintenance/Operations"
          class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
          href="/opportunities/1/virtual-internship/2/lessons/74"
        >
          <div
            class="imageWrapper"
          >
            <img
              alt="Career Pathway: Maintenance/Operations"
              class="image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <section
            class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
          >
            <h5
              class="name"
            >
              Career Pathway: Maintenance/Operations
            </h5>
          </section>
          <section
            class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
          >
            <div
              class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
            >
              Show more
              <div
                class="icon footerIcon -xs-size"
                data-testid="icon"
              >
                <div />
              </div>
            </div>
          </section>
        </a>
        <a
          aria-label="Cluster Lesson"
          class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
          href="/opportunities/1/virtual-internship/2/lessons/77"
        >
          <div
            class="imageWrapper"
          >
            <img
              alt="Cluster Lesson"
              class="image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <section
            class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
          >
            <h5
              class="name"
            >
              Cluster Lesson
            </h5>
          </section>
          <section
            class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
          >
            <div
              class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
            >
              Show more
              <div
                class="icon footerIcon -xs-size"
                data-testid="icon"
              >
                <div />
              </div>
            </div>
          </section>
        </a>
        <a
          aria-label="Project Lesson"
          class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
          href="/opportunities/1/virtual-internship/2/lessons/78"
        >
          <div
            class="imageWrapper"
          >
            <img
              alt="Project Lesson"
              class="image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0 absolute top-xs left-xs bg-success-500 rounded-full text-white p-xs"
              data-testid="icon"
            >
              <div
                class="h-sm w-sm"
              />
            </div>
          </div>
          <section
            class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
          >
            <h5
              class="name"
            >
              Project Lesson
            </h5>
          </section>
          <section
            class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
          >
            <div
              class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
            >
              Show more
              <div
                class="icon footerIcon -xs-size"
                data-testid="icon"
              >
                <div />
              </div>
            </div>
          </section>
        </a>
        <a
          aria-label="Pathway Lesson"
          class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
          href="/opportunities/1/virtual-internship/2/lessons/69"
        >
          <div
            class="imageWrapper"
          >
            <img
              alt="Pathway Lesson"
              class="image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <section
            class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
          >
            <h5
              class="name"
            >
              Pathway Lesson
            </h5>
          </section>
          <section
            class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
          >
            <div
              class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
            >
              Show more
              <div
                class="icon footerIcon -xs-size"
                data-testid="icon"
              >
                <div />
              </div>
            </div>
          </section>
        </a>
      </div>
    </div>
    <div
      class="card border border-neutral-300"
    >
      <div
        aria-labelledby="lesson-group-name"
        data-testid="lesson-group"
      >
        <h5
          class="text-sm xxxl:text-base mb-xxs xxxl:mb-xs"
          id="lesson-group-name"
        >
          Experiences
        </h5>
        <p
          class="text-xs xxxl:text-sm leading-lg mb-base xxxl:mb-md"
        >
          Gain hands-on experience  by choosing 2 products from distinct careers within this cluster.
        </p>
        <div
          class="flex flex-wrap gap-y-base gap-x-sm xxxl:gap-y-md justify-center"
        >
          <a
            aria-label="First experience opportunity"
            class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
            href="/opportunities/1/virtual-internship/2/lessons/67"
          >
            <div
              class="imageWrapper"
            >
              <img
                alt="First experience opportunity"
                class="image"
                src="https://picsum.photos/600/300"
              />
            </div>
            <section
              class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
            >
              <h5
                class="name"
              >
                First experience opportunity
              </h5>
            </section>
            <section
              class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
            >
              <div
                class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
              >
                Show more
                <div
                  class="icon footerIcon -xs-size"
                  data-testid="icon"
                >
                  <div />
                </div>
              </div>
            </section>
          </a>
          <a
            aria-label="Second experience opportunity"
            class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
            href="/opportunities/1/virtual-internship/2/lessons/68"
          >
            <div
              class="imageWrapper"
            >
              <img
                alt="Second experience opportunity"
                class="image"
                src="https://picsum.photos/600/300"
              />
            </div>
            <section
              class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
            >
              <h5
                class="name"
              >
                Second experience opportunity
              </h5>
            </section>
            <section
              class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
            >
              <div
                class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
              >
                Show more
                <div
                  class="icon footerIcon -xs-size"
                  data-testid="icon"
                >
                  <div />
                </div>
              </div>
            </section>
          </a>
        </div>
      </div>
    </div>
    <div
      aria-labelledby="lesson-group-name"
      data-testid="lesson-group"
    >
      <h5
        class="text-sm xxxl:text-base mb-xxs xxxl:mb-xs"
        id="lesson-group-name"
      >
        Post-Experience
      </h5>
      <p
        class="text-xs xxxl:text-sm leading-lg mb-base xxxl:mb-md"
      >
        Reflect on the knowledge and skills acquired and understand the next steps necessary to become successful in this career cluster.
      </p>
      <div
        class="flex flex-wrap gap-y-base gap-x-sm xxxl:gap-y-md"
      >
        <a
          aria-label="Career Review Survey"
          class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
          href="/opportunities/1/virtual-internship/2/lessons/33"
        >
          <div
            class="imageWrapper"
          >
            <img
              alt="Career Review Survey"
              class="image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <section
            class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
          >
            <h5
              class="name"
            >
              Career Review Survey
            </h5>
          </section>
          <section
            class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
          >
            <div
              class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
            >
              Show more
              <div
                class="icon footerIcon -xs-size"
                data-testid="icon"
              >
                <div />
              </div>
            </div>
          </section>
        </a>
      </div>
    </div>
  </div>
  <div
    class="card"
  >
    <div
      aria-labelledby="lesson-group-name"
      data-testid="lesson-group"
    >
      <h5
        class="text-sm xxxl:text-base mb-xxs xxxl:mb-xs"
        id="lesson-group-name"
      >
        Related Skills
      </h5>
      <p
        class="text-xs xxxl:text-sm leading-lg mb-base xxxl:mb-md"
      >
        Consider these modules to help develop skills related to this career cluster.
      </p>
      <div
        class="flex flex-wrap gap-y-base gap-x-sm xxxl:gap-y-md"
      >
        <a
          aria-label="Readiness skills section"
          class="card grow-0 shrink-0 basis-[calc(25%-theme(spacing.x))]"
          href="/opportunities/1/virtual-internship/2/lessons/58"
        >
          <div
            class="imageWrapper"
          >
            <img
              alt="Readiness skills section"
              class="image"
              src="https://picsum.photos/600/300"
            />
          </div>
          <section
            class="flex flex-col justify-between flex-1 pt-xs px-xs pb-xxs min-h-[100px]"
          >
            <h5
              class="name"
            >
              Readiness skills section
            </h5>
          </section>
          <section
            class="relative bg-primary-200 rounded-b-sm text-primary-500 before:h-xxxs before:bg-neutral-400 before:inset-0 before:absolute before:bg-primary-500"
          >
            <div
              class="flex items-center text-xxs p-xs xxxl:text-xs xxxl:py-sm xxxl:px-xs leading-lg"
            >
              Show more
              <div
                class="icon footerIcon -xs-size"
                data-testid="icon"
              >
                <div />
              </div>
            </div>
          </section>
        </a>
      </div>
    </div>
  </div>
</div>
`;
